dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java

```diff
@@ -207,17 +216,59 @@ public Response getAllMessages (
             @Context final HttpServletResponse response,
             @PathParam("language") final String language){
 
-        final long languageId = "default".equalsIgnoreCase(language) ? APILocator.getLanguageAPI().getDefaultLanguage().getId() : LanguageUtil.getLanguageId(language);
-        final Language language1 = APILocator.getLanguageAPI().getLanguage(languageId);
-        if(!UtilMethods.isSet(language1)){
-            final String message = "Language with Id or Code: "+ language + " does not exist";
-            Logger.error(this,message);
-            throw new IllegalArgumentException(message);
+        final InitDataObject initData = new WebResource.InitBuilder(webResource)
+                .requiredAnonAccess(AnonymousAccess.READ)
+                .requestAndResponse(request, response)
+                .rejectWhenNoUser(false).init();
+
+        final User user = initData.getUser();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5671019)